src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -152,9 +152,8 @@ public static bool SequenceEqual(this ReadOnlySpan<byte> first, ReadOnlySpan<byt
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool StartsWith(this Span<byte> span, ReadOnlySpan<byte> value)
         {
-            int spanLength = span.Length;
             int valueLength = value.Length;
-            return valueLength <= spanLength && SpanHelpers.StartsWith(ref span.DangerousGetPinnableReference(), spanLength, ref value.DangerousGetPinnableReference(), valueLength);
+            return valueLength <= span.Length && (valueLength == 0 || span.Slice(0, valueLength).SequenceEqual(value));
```

Comment:
You should call SpanHelpers.SequenceEqual directly - it will save the range check done by Slice.